README.md

```diff
@@ -1,75 +1,110 @@
-# PostgreSQL Trigger + Missing Index Demonstration
+# Tutorial: Simulate and fix slow delete operations in PostgreSQL
 
-This repository is an interactive demonstration of a performance pitfall common in many PostgreSQL schemas as described in [TODO - BLOG POST LINK - TODO](TODO).
+This repo contains code that lets you simulate and fix a common performance issue in PostgreSQL schemas.
 
-TL;DR: The deletion of a record that can cascade over other tables can take an unintuitive amount of time if there are no indexes on foreign key relationships **in both directions**. This repository contains scripts to create a sample schema and generate enough data to show the contrast in performance with and without these indexes.
+The issue is that deleting a single record can take an unexpectedly long time, if:
+- The record you delete is referenced as a foreign key in another table
+- Deleting this record causes rows in another table to be deleted due to an `ON DELETE CASCADE` constraint
+- You haven't created indexes on the foreign key columns in _both_ the referencing table and the referenced table.
+
+We encourage you to first read about this pitfall in [our blog post](https://render.com/blog/top-cause-slow-queries-postgresql-no-query-log-needed). Then come back here to demo it for yourself.
+
+## Overview
+In this tutorial, you will follow steps to:
+1. Create a demo environment with a demo database.
+2. Populate your database with sample data.
+3. Delete a first record. (This deletion will be slow.)
+4. Create indexes, and delete another record. (This deletion will be fast.)
 
 ## Prerequisites
 
-- `psql` command-line tool available in your system's PATH
+- Install the `psql` command-line tool on the machine where you will run the scripts in this demo. (You can run this demo on your local machine, such as a laptop.)
+
+## 1. Set up a demo environment
+### 1a. Create a demo database
+If you already have a database you can use for demo purposes, feel free to use that instead of creating a new instance. Identify the URL of this database.
+
+Otherwise, create a database on Render. You can use Render's free plan.
+1. [Follow this guide](https://docs.render.com/databases#create-your-database) to create a database through the Render dashboard.
+   * If you prefer to set up a database using Infrastructure as Code, learn about [Render Blueprints](https://docs.render.com/infrastructure-as-code) and apply the Blueprint that's [located in this repository](./render.yaml).
+2. Once your database is running, locate its [external URL](https://docs.render.com/databases#connecting-with-the-external-url) in the Render dashboard:
+
+<img width=”500” src=”./images/connection.png” />
```

Comment:
This isn't loading in the preview, not sure if it's because of the smart quotes.